app/controllers/carts_controller.rb

```diff
@@ -1,36 +1,45 @@
 class CartsController < ApplicationController
   include ActionView::Helpers::TextHelper
 
+  after_action :save_cart
```

Comment:
This should go on Orders Controller too, no?  Honestly I don't see how this hasn't been a problem so far.